### ISSUE TITLE
Any chance for a version where functions (also) return a promise?

### ISSUE THREAD

**Issue by myrne** [JamesKyburz/dialogs#33]
Title speaks for itself. Is there any chance for a version where functions (also) return a promise?

Normally I'd probably just use `Bluebird.promisifyAll`, but the non-standard callback style (i.e. not "error-first") makes this somewhat complicated.

If yes, I could submit a pull request.
If no, I'll probably create a small adapter module.

Some hybrid libraries only return a promise if no callback is provided. That could work here too.

**Comment by myrne** [comment]
If the environment has no `Promise` object, just return no promise. The user of the library would be responsible for ensuring existence of a global `Promise` object, if they want to use promises. Or allow providing it afterwards with something like `dialogs.use(Bluebird)`.

**Comment by JamesKyburz** [comment]
At present `dialogs` throws when no callback is provided except for the `alert` method, being a bit more fire and forget.

I think it would be ok to return a native promise if no cb is provided.

Somethings along the lines of

```
  if (!cb) {
    return new Promise(function (resolve, reject) {
      cb = resolve
    })
  }
```

**Comment by myrne** [comment]
Yes. It doesn't have to be more complicated than that.

BTW when I wrote my second post, I wasn't aware yet that this module exported a function, instead of an object directly. The exported function could take an extra `Promise` option, giving people the option to pass in a promise constructor if there is no native support or if they want to use a custom one. 

Maybe something like so:
```javascript
  if(!opt.Promise && typeof Promise === "function") {
    opt.Promise = Promise
  }
  if (!cb) {
    if(!opt.Promise) throw new Error("No callback and no Promise constructor available")
    return new opt.Promise(function (resolve, reject) {
      cb = resolve
    })
  }
```

**Comment by JamesKyburz** [comment]
In the case of neither opt.Promise and cb being specified I think it would be best to return a vanilla Promise (if available).

**Comment by JamesKyburz** [comment]
@braveg1rl Sorry haven't had any time to look at this. I would welcome a PR with promise support.

**Comment by JamesKyburz** [comment]
Fixed in 2.0.1.

```js
const ok = await dialogs.confirm('delete?')
```
